### PR TITLE
[MOS-578] Fix incorrect logic with SMS notifications

### PR DESCRIPTION
**Pull request by SP2FET**
Fixed case, when there's only one thread unread and there's no info about the thread, only a generic message

Tests are deleted because the fix would require rewritting ALL
of those tests - those will be written with new implementation
of notification table (MOS-899)

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [ ] Has documentation updated
- [x] Has changelog entry added

### CODE REVIEW

File: module-db/Interface/NotificationsRecord.hpp [hunk 1]
```diff
@@ -84,6 +84,7 @@ class NotificationsRecordInterface : public RecordInterface<NotificationsRecord,
   private:
     NotificationsDB *notificationsDb   = nullptr;
     ContactRecordInterface *contactsDb = nullptr;
+    std::vector<uint32_t> unreadMsgNumberIDs;
```
Comment (Maciej-Mudita) [inline module-db/Interface/NotificationsRecord.hpp:87]:
```suggestion
    std::vector<std::uint32_t> unreadMsgNumberIDs;
```

File: module-db/Interface/NotificationsRecord.hpp [hunk 2]
```diff
@@ -98,5 +99,5 @@ class NotificationsRecordInterface : public RecordInterface<NotificationsRecord,
     [[nodiscard]] bool processIncrement(NotificationsRecord::Key key,
                                         std::optional<utils::PhoneNumber::View> &&number,
                                         size_t size);
-    [[nodiscard]] bool processDecrement(NotificationsRecord::Key key, size_t size);
+    [[nodiscard]] bool processDecrement(NotificationsRecord::Key key, uint32_t numberID, size_t size);
```
Comment (Maciej-Mudita) [inline module-db/Interface/NotificationsRecord.hpp:102]:
```suggestion
    [[nodiscard]] bool processDecrement(NotificationsRecord::Key key, std::uint32_t numberID, size_t size);
```

File: module-db/Interface/NotificationsRecord.cpp [hunk 3]
```diff
@@ -256,17 +261,32 @@ bool NotificationsRecordInterface::processIncrement(NotificationsRecord::Key key
     return ret;
 }
 
-bool NotificationsRecordInterface::processDecrement(NotificationsRecord::Key key, size_t delta)
+bool NotificationsRecordInterface::processDecrement(const NotificationsRecord::Key key,
+                                                    const uint32_t numberID,
```
Comment (Maciej-Mudita) [inline module-db/Interface/NotificationsRecord.cpp:265]:
```suggestion
                                                    const std::uint32_t numberID,
```

**Review by Maciej-Mudita — APPROVED** [review]

File: module-db/Interface/NotificationsRecord.hpp (hunk 1 above)
Comment (SP2FET) [reply module-db/Interface/NotificationsRecord.hpp]: fixed

File: module-db/Interface/NotificationsRecord.hpp (hunk 2 above)
Comment (SP2FET) [reply module-db/Interface/NotificationsRecord.hpp]: fixed

File: module-db/Interface/NotificationsRecord.cpp (hunk 3 above)
Comment (SP2FET) [reply module-db/Interface/NotificationsRecord.cpp]: fixed

**Review by Lefucjusz — APPROVED** [review]

**Pull request merged**